marie/ocr/ocr_engine.py

```diff
@@ -253,114 +254,136 @@ def __process_extract_regions(
+                    if crop_to_content_enabled:
+                        region_fragment = crop_to_content(region_fragment)
+                        h = region_fragment.shape[0]
+                        w = region_fragment.shape[1]
+                        padding = 4
+
+                    if padding != 0:
+                        region_overlay = (
+                            np.ones(
+                                (h + padding * 2, w + padding * 2, 3), dtype=np.uint8
+                            )
+                            * 255
+                        )
+                        region_overlay[
+                            padding : h + padding, padding : w + padding
+                        ] = region_fragment
+                    else:
+                        region_overlay = region_fragment
 
-                del boxes
-                del img_fragments
-                del lines
-                del lines_bboxes
-
-                if not filter_snippets:
-                    result["overlay_b64"] = encodeToBase64(overlay_image)
-
-                result["id"] = rid
-                extended.append(result)
-
-                # TODO : Implement rendering modes
-                # 1 - Simple
-                # 2 - Full
-                # 3 - HOCR
-                self.logger.debug(result)
-                rendering_mode = "simple"
-                region_result = {}
-                if rendering_mode == "simple":
-                    if "lines" in result and len(result["lines"]) > 0:
-                        lines = result["lines"]
-                        line = lines[0]
-                        region_result["id"] = rid
-                        region_result["text"] = line["text"]
-                        region_result["confidence"] = line["confidence"]
-                        output.append(region_result)
+                    # cv2.imwrite(f"/tmp/marie/region_overlay_{page_index}_{rid}.png", region_overlay)
+                    # each region can have its own segmentation mode
+                    if "mode" in region:
+                        mode = PSMode.from_value(region["mode"])
                     else:
+                        mode = pms_mode
+
+                    # create cache key from region id and overlay hash
+                    region_overlay_hash = hash_frames_fast(region_overlay)
+                    cache_key = f"{id(region)}_{region_overlay_hash}"
+                    bbox_results = None
+
+                    if cache_key in bbox_cache:
+                        bbox_results = bbox_cache[cache_key]
+
+                    if bbox_results is None:
+                        bbox_results = box_processor.extract_bounding_boxes(
+                            queue_id, checksum, region_overlay, psm=mode
+                        )
+                        bbox_cache[cache_key] = bbox_results
+
+                    bbox_results_batch.append(bbox_results)
+
+                except Exception as ex:
```

Comment:
This try:except should be moved to to encompass at least to line 356. This allows us to log any errors arising from either the box_processor or the icr_processor